diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt

```diff
@@ -283,11 +349,80 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
             is LongBinaryExpression -> fixLongBinaryExpression(fixableType)
             is BinaryExpression -> fixBinaryExpression(fixableType.node)
             is StringTemplate -> fixStringTemplate(fixableType)
+            is DotQualifiedExpression -> fixDotQualifiedExpression(fixableType)
+            is ValueArgumentList -> fixArgumentList(fixableType)
             is Lambda -> fixLambda(fixableType.node)
+            is WhenEntry -> fixWhenEntry(fixableType)
             is None -> return
         }
     }
 
+    private fun fixWhenEntry(wrongWhenEntry: WhenEntry) {
+        val node = wrongWhenEntry.node
+        node.getFirstChildWithType(ARROW)?.let {
+            node.addChild(PsiWhiteSpaceImpl(" "), it.treeNext)
+            node.addChild(PsiWhiteSpaceImpl("{"), it.treeNext.treeNext)
+            node.appendNewlineMergingWhiteSpace(it.treeNext.treeNext.treeNext, it.treeNext.treeNext.treeNext)
+        }
+        node.appendNewlineMergingWhiteSpace(node.lastChildNode.treeNext, node.lastChildNode.treeNext)
+        node.addChild(PsiWhiteSpaceImpl("}"), node.lastChildNode.treeNext)
+    }
+
+    private fun fixDotQualifiedExpression(wrongDotQualifiedExpression: DotQualifiedExpression) {
+        val node = wrongDotQualifiedExpression.node
+        val dot = node.getFirstChildWithType(DOT)
+        node.appendNewlineMergingWhiteSpace(dot, dot)
+    }
+
+    private fun fixArgumentsListFirstArgument(wrongArgumentList: ValueArgumentList): Int {
+        val lineLength = wrongArgumentList.maximumLineLength.lineLength
+        val node = wrongArgumentList.node
+        var offset = 0
+        node.getFirstChildWithType(COMMA)?.let {
+            if (positionByOffset(it.startOffset).second > lineLength) {
+                node.appendNewlineMergingWhiteSpace(node.findChildByType(LPAR)!!.treeNext, node.findChildByType(LPAR)!!.treeNext)
+                node.appendNewlineMergingWhiteSpace(node.findChildByType(RPAR), node.findChildByType(RPAR))
+                offset = 50
+            }
+        } ?: node.getFirstChildWithType(RPAR)?.let {
+            node.appendNewlineMergingWhiteSpace(node.findChildByType(LPAR)!!.treeNext, node.findChildByType(LPAR)!!.treeNext)
+            node.appendNewlineMergingWhiteSpace(node.findChildByType(RPAR), node.findChildByType(RPAR))
+            offset = 50
+        }
+        return offset
+    }
+
+    /**
+     * Fix arguments in arguments list
+     */
+    private fun fixArgumentList(wrongArgumentList: ValueArgumentList) {
+        val lineLength = wrongArgumentList.maximumLineLength.lineLength
+        val node = wrongArgumentList.node
+        val offset = fixArgumentsListFirstArgument(wrongArgumentList)
+        val listComma = node.getAllChildrenWithType(COMMA).map {
+            it to positionByOffset(it.startOffset - offset).second
+        }.sortedBy { it.second }
+        var lineNumber = 1
+        listComma.forEachIndexed { index, pair ->
```

Comment:
@akuleshov7, @petertrr, @nulls, please have a look.

Reply:
do you want to avoid `var`?
if yes, fold can be used here:
```kotlin
val lineNumber = listComma.foldIndexed(1) { index, result, pair ->
       if (pair.second >= result * lineLength) {
            val commaSplit = if (index > 0) {
                listComma[index - 1].first
            } else {
                pair.first
            }
            node.appendNewlineMergingWhiteSpace(commaSplit.treeNext, commaSplit.treeNext)
            result + 1
        } else {
            result
        }
}
```